#Longest Increasing Subsequence in Brute Foce

In [24]:
import datetime
import random
import numpy as np

In [25]:
def LIS(arr):
    l = []
    for i in range(0,len(arr)):
        n = arr[i]
        temp = [n]
        for j in range(i+1,len(arr)):
            if arr[j] > n:
                temp.append(arr[j])
                n=arr[j]
        l.append(temp)

    max = 0
    for k in l:
        if len(k) > max:
            max = len(k)



In [26]:
def empericalAnalysis():
    i = 0
    time=[]
    while i< 50:
        array = np.random.randint(1,9999999,1000)
        start_time =  datetime.datetime.now()

        LIS(array)    
        
        end_time =  datetime.datetime.now()
        time_elapsed = (end_time - start_time)
        time.append(int(time_elapsed.total_seconds() * 1000))
        i=i+1
    print(sum(time) / len(time))

In [27]:
def main():
    arr =[50,3,10,7,40,80]
    #[10, 22, 9, 33, 21, 50, 41, 60]
    #[10, 22, 9, 33, 21, 50, 41, 60, 80] 
    #[10, 5, 8, 3, 9, 4, 12, 11]
    LIS(arr)
    #empericalAnalysis()

In [28]:
if __name__ == "__main__":
    main()

58.34


avarage time for 50 runs of array of size 1000 is 58.34